PostFreeSurfer/PostFreeSurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,420 @@
+
+opts_AddMandatory '--path' 'StudyFolder' 'path' "folder containing all timepoins and templates" "--path"
+opts_AddMandatory '--subject'   'Subject'   'subject ID' "Subject ID. Note: this is distinct subject ID. It is to be used \
+                    in the output folder name as <Subject>.long.<Template> for the template and <Subject>.long.<Session> for timepoint"
+opts_AddMandatory '--longitudinal-template'  'Template'  'FS template ID' "Longitudinal template ID (same as Freesurfer long template ID)"
+opts_AddMandatory '--sessions' 'Timepoints_string' 'FS timepoint ID(s)' "Freesurfer timepoint ID(s). For timepoint (session)\
+                    processing, specify current timepoint. For template processing, must specify all timepoints, @ separated.\
+                    Timepoint ID and Session are synonyms in HCP structural pipelines."
+opts_AddMandatory '--template_processing' 'TemplateProcessing' 'create template flag' "0 if TP processing; 1 if template processing (must be run after all TP's)"
+
+#the following options have the same meaning as in PreFreesurferPipeline.
+opts_AddMandatory '--t1template' 'T1wTemplate' 'file_path' "MNI T1w template"
+opts_AddMandatory '--t1templatebrain' 'T1wTemplateBrain' 'file_path' "Brain extracted MNI T1wTemplate"
+opts_AddMandatory '--t1template2mm' 'T1wTemplate2mm' 'file_path' "MNI 2mm T1wTemplate"
+opts_AddMandatory '--t2template' 'T2wTemplate' 'file_path' "MNI T2w template"
+opts_AddMandatory '--t2templatebrain' 'T2wTemplateBrain' 'file_path' "Brain extracted MNI T2wTemplate"
+opts_AddMandatory '--t2template2mm' 'T2wTemplate2mm' 'file_path' "MNI 2mm T2wTemplate"
+opts_AddMandatory '--templatemask' 'TemplateMask' 'file_path' "Brain mask MNI Template"
+opts_AddMandatory '--template2mmmask' 'Template2mmMask' 'file_path' "Brain mask MNI 2mm Template"
+opts_AddMandatory '--fnirtconfig' 'FNIRTConfig' 'file_path' "FNIRT 2mm T1w Configuration file"
+
+#needed to transform Freesurfer mask to MNI space.
+opts_AddMandatory '--freesurferlabels' 'FreeSurferLabels' 'file' "location of FreeSurferAllLut.txt"
+
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var FSLDIR
+log_Check_Env_Var HCPPIPEDIR_Global
+
+# Naming Conventions
+
+T1wImageBrainMask="brainmask_fs"
+MNI_hires_template="$HCPPIPEDIR/global/templates/MNI152_T1_0.8mm.nii.gz"
+TemplateProcessing=$(opts_StringToBool "$TemplateProcessing")
+
+
+if [[ "$Timepoints_string" =~ "@" ]]; then 
+    if (( TemplateProcessing )); then    
+            log_Err_Abort "More than one timepoint is specified in template mode, please check calling script."
+    fi
+    IFS='@' read -r -a timepoints <<< "$Timepoints_string"
+    Timepoint_cross=${timepoints[0]}
+else
```

Comment:
```suggestion
if [[ "$Timepoints_string" =~ "@" ]]; then
    if (( TemplateProcessing )); then
        log_Err_Abort "More than one timepoint is specified in template mode, please check calling script."
    fi
    IFS='@' read -r -a timepoints <<< "$Timepoints_string"
    #Timepoint_cross is only used in variables that get used in non-template mode, but give it a value to avoid "unset variable" errors
    Timepoint_cross=${timepoints[0]}
else
```
Ideally, you could move all variables specific to non-template mode into a conditional that only gets executed in non-template mode, but a comment like this may be an acceptable substitute.  I also tend to separate by processing mode first, and then sanity check the inputs for the mode, but I suppose doing it this way isn't too confusing yet...

Also, indentation (and stray spaces on the end of lines).

Reply:
Ok